src/lib/DALMeteorMethods/UserMethods.js

```diff
@@ -1,19 +1,104 @@
 /*eslint-disable*/
 Users = new Mongo.Collection.get("users");
 if (Meteor.isServer) {
-    Meteor.publish("users", function(){
-      return Users.find({_id:this.userId});
+    Meteor.publish("users", function() {
+        return Users.find({
+            _id: this.userId
+        });
     });
-}else {
-  Meteor.subscribe("users");
+    Meteor.publish("usersInRoom", function() {
+        return Users.find({});
+    });
+} else {
+    Meteor.subscribe("users");
+    Meteor.subscribe("usersInRoom");
 }
 UserMethods = {};
 
-UserMethods.getUserImage = function(user){
-  var userImage = Users.findOne({_id:user}).services.google.picture;
-  return userImage;
+UserMethods.getUserImage = function(user) {
+    var userImage = Users.findOne({
+        _id: user
+    }).services.google.picture;
+    return userImage;
+};
+
+UserMethods.getAuthor = function() {
+    return Meteor.user() ? Meteor.user().profile.name : Session.get("author");
 };
 
-UserMethods.getAuthor = function(){
-  return Meteor.user() ? Meteor.user().profile.name:Session.get("author");
+UserMethods.getUsersInRoom = function(roomCode) {
+    return Users.find({
+        "roomCodes": {
+            $in: [roomCode]
+        }
+    });
+};
+
+UserMethods.addUserToRoom = function(author, roomCode) {
+    if (Meteor.user()) {
+        var user = Users.findOne({
+            _id: Meteor.userId()
+        });
+        var roomCodes = user.roomCodes;
+        if (roomCodes === undefined)
+            roomCodes = [roomCode];
+        else if (roomCodes.indexOf(roomCode) === -1) {
+            roomCodes.push(roomCode);
+        }
+        Users.update({
+            _id: Meteor.userId()
+        }, {
+            $set: {
+                roomCodes: roomCodes
+            }
+        });
+    } else {
```

Comment:
I actually really like this, making the non signed in users data look more similar to the signed in user data, nice!